### DISCUSSION TITLE
Problems with DDoS attacks

### DISCUSSION THREAD

**Discussion by OpenSource03** [pterodactyl/panel#3142] [Support] [answered]
Hello there!

While under DDoS, the website is consuming 100% of 4 core Ryzen 7 VPS, while only about 10Mbit is bypassing. Is there anyone with a solution on how to solve it? I've tried multiple PHP-FPM configurations, but never managed to get any result.

**Comment by OpenSource03** [comment]
![Screen Shot 2021-03-04 at 11 10 26 PM](https://user-images.githubusercontent.com/29690431/110037566-e5b1a100-7d3e-11eb-8ac8-502b8bdfdb26.png)

**Reply by Slitzo3** [reply]
I don't think it's a DDOS attack. Just someone overloading your webserver. So just block the ip where it's coming from.

**Reply by OpenSource03** [reply]
It is. Anonboot stress test. I performed it myself to demonstrate the effect of it.

**Comment by schrej** [comment]
Can you check you webserver access logs and see which endpoints they are calling? If it's actually very little traffic, it might be some kind of DoS vulnerability in the panel. 

I also don't see redis or memcached, do you have caching set up? That probably helps _a lot_ with performance.

**Reply by OpenSource03** [reply]
Most usually it is GET /auth/login. The problem is that laravel is always calling MySQL on every single request adding a row or updating session table rows. It is producing tremendous load on php and MySQL from what I can see? Also, would you mind explaining how to enable such caching mechanics?

Thanks!

**Reply by OpenSource03** [reply]
Figured it out. Thanks, it really helped a lot!

**Comment by OpenSource03** [comment, marked as answer]
**SOLUTION FOR ANYONE WITH SIMILAR PROBLEMS**

1. Execute `sudo apt-get install php7.4-memcached`
2. Go to your Pterodactyl folder and execute `php artisan p:environment:setup`
3. Configure your session driver and cache driver as memcached
